src/providers/aws.rs

```diff
@@ -748,23 +763,45 @@ impl RegionLauncher {
                 .wrap_err("failed to make spot instance requests")?;
 
                 let start = time::Instant::now();
-                self.wait_for_spot_instance_requests(max_wait)
+                if let Err(e) = self
+                    .wait_for_spot_instance_requests(max_wait)
                     .await
-                    .wrap_err("failed while waiting for spot instances fulfilment")?;
+                    .wrap_err(eyre!(
+                        "failed while waiting for spot instances fulfilment in {}",
+                        self.region.name()
+                    ))
+                {
+                    // if wait_for_spot_instance_requests returned an Err, it will have cleaned up
+                    // the spot instance requests already.
+                    if let LaunchMode::TrySpot { .. } = mode {
+                        tracing::debug!(err = ?e, "re-trying with OnDemand instace");
+                        do_ondemand = true;
```

Comment:
Hmm, this won't work. The `Err(e)?` is still executed, which will return early. I think you need an `else` here?

Reply:
yeah, good catch :+1: